TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -102,13 +102,13 @@ public override bool Visit(Move move)
                 DataDefinition fromVariable = null;
                 DataDefinition toVariable = null;
                 //For MoveCorrespondingStatement check children compatibility
-                if (move.StorageAreaReadsDataDefinition.TryGetValue(moveCorresponding.FromGroupItem,
+                if (move.StorageAreaReadsDataDefinition!=null && move.StorageAreaReadsDataDefinition.TryGetValue(moveCorresponding.FromGroupItem,
```

Comment:
Why check for null here?
StorageAreaReadsDataDefinition must always be populated by CheckVariable before we enter this code.

Reply:
I changed the place where the dictionaries are initialized. In CheckVariable only when a single DataDefintion is found the dictionaries are initialized, if they are null. 
In the case where a variable is not defined (does not have a DataDefintion) in CheckVariable the dictionary will not be initialized anymore, so a check for null in the Visit(Move mode) method is needed.
I do think that this is a better choice, as dictionaries will be initialized only where needed, so memory consumption is lowered.

Reply:
Then can you force the use of method Node.GetDataDefinitionFromStorageAreaDictionary instead of direct access to StorageAreaReadsDataDefinition and StorageAreaWritesDataDefinition.

This will avoid to remember to check for null everywhere in the code.

Reply:
Not quite. Node.GetDataDefinitionFromStorageAreaDictionary will search in both StorageAreaReadsDataDefinition and StorageAreaWritesDataDefinition dictionaries. 
In Visit(Move mode) method we need to extract the to and from members which come from either StorageAreaReadsDataDefinition (from member) or StorageAreaWritesDataDefinition (to memeber). Thus we cannot use GetDataDefinitionFromStorageAreaDictionary  because we do not know from what dictionary the retuned DataDefinition comes.

For now, in Visit(Move mode) is the only place where a null check was needed before searching a value in the StorageArea dictionaries. 

If you want to avoid null checks in future, I can move the dictionary initialization where it was before in CheckVariable method. However, in my opinion, this will degrade more the performance than making a null check where is needed, as it will allocate memory for 2 dictionaries (and they may not be used).